mlflow/deployments/server/app.py

```diff
@@ -330,6 +330,65 @@ async def get_limits(endpoint: str) -> LimitsConfig:
     async def set_limits(payload: SetLimitsModel) -> LimitsConfig:
         raise HTTPException(status_code=501, detail="The set_limits API is not available yet.")
 
+    def _look_up_route(name: str) -> Optional[Route]:
+        if r := app.dynamic_routes.get(name):
+            return r
+
+        raise HTTPException(
+            status_code=404,
+            detail=f"Route {name} not found in the configuration.",
+        )
+
+    # OpenAI compatible endpoints
+    @app.post("/v1/chat/completions")
+    async def chat_handler(request: Request, payload: chat.RequestPayload) -> chat.ResponsePayload:
+        route = _look_up_route(payload.model)
+        if route.route_type != RouteType.LLM_V1_CHAT:
+            raise HTTPException(
+                status_code=404,
+                detail=f"Route {route.name} is not a chat route.",
+            )
+
+        prov = get_provider(route.model.provider)(route)
+        payload.model = None
```

Comment:
Can we also add inline comments explaining why we reset the model in the payload to `None`?